ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/results/tracing/Trace.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.jsonrpc.internal.results.tracing;
+
+public interface Trace {
+  @FunctionalInterface
+  interface ResultWriter {
```

Comment:
Style: why an interface in an interface?  I would expect just one interface `TraceResultWriter`

Reply:
`Trace` interface is the common type for `FlatTrace` and the other types of traces we will add for `vmTrace` and `stateDiff` options. So i'll keep `Trace` interface but as you suggest i'll move `ResultWriter`to a separate file.

Reply:
Does that make sense ?

Reply:
Done.

Reply:
Totally agree. i'll do that

Reply:
Done.